src/root.component.tsx

```diff
@@ -1,9 +1,26 @@
 import React from "react";
 import openmrsRootDecorator from "@openmrs/react-root-decorator";
 import styles from "./root.styles.css";
+import authResource from "./auth.resource";
+import { toggle } from "kremling";
+import { BrowserRouter, Redirect } from "react-router-dom";
 
-function Root() {
+function Root(props: NavProps) {
   const [sidenavOpen, setSidenavOpen] = React.useState(false);
+  const [isUserMenuOpen, setIsUserMenuOpen] = React.useState(false);
+  const [userNames, setUserNames] = React.useState(null);
+  const [userInitials, setUserInitials] = React.useState(null);
+  const [isLoggedIn, setIsLoggedIn] = React.useState(true);
+
+  React.useEffect(() => {
+    authResource.getLoggedInUser().subscribe(({ person }) => {
```

Comment:
We should store the subscription and unsubscribe during the effect's cleanup function. See https://reactjs.org/docs/hooks-effect.html#effects-with-cleanup

const subscription = authResource.getLoggedInUser().subscribe(...)
return () => subscription.unsubscribe()

Reply:
Oops I keep forgetting to do that